distributed/client.py

```diff
@@ -602,9 +602,29 @@ def set_error(self, exception, traceback):
             The exception
         traceback: Exception
             The traceback
+        erred_on: set
+            Workers where it errored
+        key: tuple
+            Task key
         """
         _, exception, traceback = clean_exception(exception, traceback)
 
+        if (
+            dask.config.get("admin.traceback.shorten.when", None)
+            and dask.config.get("admin.traceback.shorten.what", None)
+            and key
+            and erred_on
+        ):
+            # when Exception is logged or printed, its args are printed as well,
+            # this makes it easier to communicate extra information without
+            # wrapping it with a custom Exception
+            worker_message = f"in task: {key}\non worker: {erred_on}"
+            if not exception.args:
+                exception.args = (worker_message,)
+            else:
+                arg0 = f"{exception.args[0]}\n{worker_message}"
+                exception.args = (arg0,) + exception.args[1:]
```

Comment:
What kind of exception is this? is this a *specific* exception type raised by distributed?
args can contain very different things depending on the exception subclass and this change will break many custom exceptions.

Reply:
This comment has not been addressed.
In fact, I can very easily make it break:

```python
class BadNumbers(Exception):
    def __str__(self):
        return "I didn't like these numbers: " + " ".join(f"{x:.1f}" for x in self.args)

def f():
    raise BadNumbers(1, 2.5, 3)
    
f()

---------------------------------------------------------------------------
BadNumbers                                Traceback (most recent call last)
Cell In[10], line 8
      5 def f():
      6     raise BadNumbers(1, 2.5, 3)
----> 8 f()

Cell In[10], line 6, in f()
      5 def f():
----> 6     raise BadNumbers(1, 2.5, 3)

BadNumbers: I didn't like these numbers: 1.0 2.5 3.0

import distributed
client = distributed.Client()
client.submit(f).result()

Custom TB Handler failed, unregistering
---------------------------------------------------------------------------
BadNumbers                                Traceback (most recent call last)
    [... skipping hidden 1 frame]

Cell In[12], line 1
----> 1 c.submit(f).result()

Cell In[10], line 6, in f()
      5 def f():
----> 6     raise BadNumbers(1, 2.5, 3)

<class 'str'>: (<class 'ValueError'>, ValueError("Unknown format code 'f' for object of type 'str'"))
[...]
TypeError: structured_traceback() missing 1 required positional argument: 'evalue'
```

Reply:
Ah. I addressed this comment here:

https://github.com/dask/distributed/pull/7847#discussion_r1237787343

But I changed the implementation since, so it's not relevant anymore.